thrift/src/main/java/com/linecorp/armeria/common/thrift/text/StackWalkingCurrentThriftMessageClassFinder.java

```diff
@@ -0,0 +1,98 @@
+// Copyright 2011 Twitter, Inc.
+// -------------------------------------------------------------------------------------------------
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this work except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file, or at:
+//
+//  https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =================================================================================================
+package com.linecorp.armeria.common.thrift.text;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class StackWalkingCurrentThriftMessageClassFinder extends AbstractCurrentThriftMessageClassFinder {
+    private final Logger logger =
+            LoggerFactory.getLogger(StackWalkingCurrentThriftMessageClassFinder.class);
+
+    private final MethodHandle walkMH;
+    private final MethodHandle getInstanceMH;
+    private MethodHandle getClassNameMH;
+
+    StackWalkingCurrentThriftMessageClassFinder() throws Exception {
+        final ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+        final MethodHandles.Lookup lookup = MethodHandles.lookup();
+        final MethodType walkMT = MethodType.methodType(Object.class, Function.class);
+        final Class<?> stackWalker = classLoader.loadClass("java.lang.StackWalker");
+        walkMH = lookup.findVirtual(stackWalker, "walk", walkMT);
+
+        final MethodType getInstanceMT = MethodType.methodType(stackWalker);
+        getInstanceMH = lookup.findStatic(stackWalker, "getInstance", getInstanceMT);
+
+        final Class<?> stackFrame = classLoader.loadClass("java.lang.StackWalker$StackFrame");
+        final MethodType getClassNameMT = MethodType.methodType(String.class);
+        getClassNameMH = lookup.findVirtual(stackFrame, "getClassName", getClassNameMT);
+    }
+
+    @Nullable
+    private final Function<Stream<Object>, Class<?>> walkHandler = s ->
+            s.map(stackFrame -> {
+                try {
+                    return getClassByName(getClassNameMH.invoke(stackFrame).toString());
+                } catch (Throwable t) {
+                    logger.warn("Failed to invoke StackWalker.StackFrame.getDeclaringClass():", t);
+                }
+
+                return null;
+            })
+            .map(AbstractCurrentThriftMessageClassFinder::getMatchedClass)
```

Comment:
How about merging this map() with the map() above for efficiency?

    returb getMatchedClass(getClassByName(...));